ios/RNSensitiveInfo/RNSensitiveInfo.m

```diff
@@ -209,8 +209,30 @@ - (NSString *)messageForError:(NSError *)error
                                   }
                                   return;
                               }
-                              
-                              [self getItemWithQuery:query resolver:resolve rejecter:reject];
+            
+                           // NOTE: a workaround where we load the last domain state from touch id
+                           // to determine if the biomtric enrollment had any changes on the device
+                           NSUserDefaults *defaults = [NSUserDefaults standardUserDefaults];
+                           NSData *oldDomainState = [defaults objectForKey:@"domainTouchID"];
+                           NSData *domainState = [context evaluatedPolicyDomainState];
+            
+                           // check for domain state changes and not nil (in case of pincode fallback or fresh biometric setup)
+                           if (![oldDomainState isEqual:domainState] && domainState != nil && oldDomainState != nil) {
```

Comment:
This code is used on iOS to detect if the user changed his biometric setup. We run that every time a token is read, so in case it changes we delete the previously stored token and reject. The key is `domainTouchID` but it also works for `Face ID`. However because the we are using `localizedFallbackTitle` the lowest security barrier is the users pin code. (and we want to support pin fallback as a UX requirement).

This will only reject and delete the token after the user provides his biomtric data and iOS detects that it is not the same as the one stored in `domainState` via `LAContext`